lens/private/list/append.rkt

```diff
@@ -2,81 +2,17 @@
 
 (provide append*-lens append*n-lens)
 
-(require lens racket/match racket/list)
+(require "flatten.rkt")
 
 (module+ test
-  (require rackunit syntax/parse lens/private/test-util/test-lens))
-
-;; restore-structure : (Listof (Listof A)) (Listof B) -> (Listof (Listof B))
-;; Takes a list of lists and a list and un-flattens the flattened
-;; argument according to the structure of the structure arguement.
-;; The length of the flattened list must be the same as the length
-;; of (append* structure).
-(define (restore-structure structure flattened)
-  (match structure
-    [(list)
-     (unless (empty? flattened)
-       (error 'append*-lens "flattened list is too long to match the structure"))
-     structure]
-    [(cons s-lst s-rst)
-     (define-values [f-lst f-rst]
-       (split-at flattened (length s-lst)))
-     (cons f-lst (restore-structure s-rst f-rst))]))
-
-;; append*-lens : (Lens (Listof (Listof Any)) (Listof Any))
-;; where the only valid views are lists with the same length as the
-;; result of applying append* to the target.
-(define append*-lens
-  (make-lens
-   append*
-   restore-structure))
+  (require rackunit lens lens/private/test-util/test-lens))
 
 (define (append*n-lens n)
-  (apply lens-thrush (make-list n append*-lens)))
+  (flatten/depth-lens (add1 n)))
```

Comment:
Drat, forgot about that (maybe I should make `define/compose` in point-free do a static rename). Nevermind then.